checkout/views.py

```diff
@@ -0,0 +1,190 @@
+from django.shortcuts import render, redirect, reverse
+from django.conf import settings
+from django.contrib import messages
+from django.core.mail import send_mail
+from django.template.loader import render_to_string
+
+from .forms import OrderForm
+from .models import Order, OrderLineItem
+from products.models import Product
+from profiles.models import UserProfile
+from profiles.forms import UserProfileForm
+
+
+def checkout(request):
+
+	if request.method == 'POST':
+		cart = request.session.get('cart', {})
+
+		form_data = {
+			'full_name': request.POST['full_name'],
+			'email': request.POST['email'],
+			'phone_number': request.POST['phone_number'],
+			'country': request.POST['country'],
+			'postcode': request.POST['postcode'],
+			'town_or_city': request.POST['town_or_city'],
+			'street_address1': request.POST['street_address1'],
+			'street_address2': request.POST['street_address2'],
+			'county': request.POST['county'],
+		}
+
+		order_form = OrderForm(form_data)
+		if order_form.is_valid():
+			order = order_form.save()
+			for item_id, item_data in cart.items():
+				try:
+					product = Product.objects.get(id=item_id)
+					# If we're working w/ an item with no sizes, item_data will be the quantity
+					if isinstance(item_data, int):
+						order_line_item = OrderLineItem(
+							order = order, 
+							product = product, 
+							quantity = item_data
+						)
+						order_line_item.save()
+					# Otherwise we need to add order lineitems for each size
+					else:
+						for size, quantity in item_data['items_by_size'].items():
```

Comment:
This is getting very deeply nested, I'd suggest refactoring to do some of this functionality in model methods

Reply:
Yeah, I agree. I'll see what I can do to extract some of this and clean it up!